spec/parser_spec.lua

```diff
@@ -419,6 +419,19 @@ describe("Parser /", function()
             ]], "local function name has a ':'")
         end)
 
+        it("disallow too many function parameters", function ()
+           local params = ''
+           for i = 1, 200 do
+              params = params..'a'..i..', '
+           end
+           params = params..'a201'
```

Comment:
It's better to avoid repeated string concatenation inside a loop, that's O(n^2)  for Lua. I suggest using table.concat, which should also allow you to stop special-casing the last arguments.

Also, our coding style favors double-quoted strings.

```lua
local xs = {}
for i = 1, 201 do
    xs[i] = "a"..i
end
local params = table.concat(xs, ", ")
```